### PR TITLE
Clean code and upgrade deps

### PR DESCRIPTION
**Pull request by chaoticlonghair**

### CODE REVIEW

File: src/core/cita_bft.rs [hunk 1]
```diff
@@ -50,24 +50,40 @@ const INIT_ROUND: usize = 0;
...
-
 pub type TransType = (String, Vec<u8>);
 pub type PubType = (String, Vec<u8>);
 
+#[derive(Debug, PartialEq, PartialOrd, Clone, Copy)]
+enum VerifiedProposalStatus {
+    Ok,
+    Failed,
+    Undo,
+}
```
Comment (u2) [inline src/core/cita_bft.rs]:
```
enum Foo {
   A = 1,
   B = -1,
   C = 0, 
}
```

File: src/core/cita_bft.rs [hunk 2]
```diff
@@ -50,24 +50,40 @@ const INIT_ROUND: usize = 0;
...
+    fn from(st: VerifiedProposalStatus) -> i8 {
+        match st {
+            VerifiedProposalStatus::Ok => 1,
+            VerifiedProposalStatus::Failed => -1,
+            VerifiedProposalStatus::Undo => 0,
+        }
+    }
+}
+
+impl From<i8> for VerifiedProposalStatus {
```
Comment (u2) [inline src/core/cita_bft.rs:66]: https://www.enodev.fr/posts/rusticity-convert-an-integer-to-an-enum.html

https://github.com/rust-num/num-derive/blob/master/tests/with_custom_values.rs

**Review by u2 — CHANGES_REQUESTED** [review]

File: src/core/cita_bft.rs (hunk 1 above)
Comment (chaoticlonghair) [reply src/core/cita_bft.rs]: Is not `i8`.

Comment (u2) [reply src/core/cita_bft.rs]: https://github.com/cryptape/cita-bft/blob/develop/src/core/wal.rs#L140

Convert it to fixed-sized u8 array and then saving. The type would not cause compatibility problem.

Comment (chaoticlonghair) [reply src/core/cita_bft.rs]: There are two enum, `u8` is another one, this is `i8`.

**Review by u2 — APPROVED** [review]

**Pull request merged**